src/reflect/scala/reflect/internal/SymbolTable.scala

```diff
@@ -171,11 +171,12 @@ abstract class SymbolTable extends macros.Universe
   final val NoRunId = 0
 
   // sigh, this has to be public or enteringPhase doesn't inline.
-  var phStack: List[Phase] = Nil
+  var phStack: Array[Phase] = new Array(128)
```

Comment:
Storage wise, there isn't a difference, both are just already allocated `Object` refs.

If we store the phase name, we have to eagerly call the virtual `Phase.name` in `phase_=`, here, even though under normal circumstances we don't use the result.

In general, It would be good to avoid retaining `Phase`-s as fields in `Global` so are eligible for GC after the `Run` that spawned them has been collected. But we know we'll end up with an empty phase stack here at the end of the run, so it doesn't make a practical difference.